src_python/fl_client.py

```diff
@@ -0,0 +1,223 @@
+
+                
+                logging.info('Training started')
+                self.train()
+                logging.info('Training done')
+
+                # eval = self.MODEL.evaluate()
+
+                # f1_train = (2 * eval[0][2] * eval[0][4]) / (eval[0][2] + eval[0][4])
+                # f1_test = (2 * eval[1][2] * eval[1][4]) / (eval[1][2] + eval[1][4])
+                # logging.info('After Round %s - Local model - Training set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[0][0], eval[0][1],eval[0][2],eval[0][3],f1_train,eval[0][4])
+                # logging.info('After Round %s - Local model - Testing set evaluation : loss - %s, accuracy - %s, recall - %s, AUC - %s, F1 - %s, precision - %s',self.rounds, eval[1][0], eval[1][1],eval[1][2],eval[1][3],f1_test,eval[1][4])
+
+                logging.info('Sent local model to the server')
+                self.send_model()
+
+
+if __name__ == "__main__":
+
+    from models.supervised import Model
+
+    if 'IP' not in args.keys()  or args['IP'] == 'localhost':
+        args['IP'] = socket.gethostname()
+
+    if 'PORT' not in args.keys():
+        args['PORT'] = 5000
+
+    if 'epochs' not in args.keys():
+        args['epoch'] = 10
+
+    logging.warning('####################################### New Training Session #######################################')
+    logging.info('Client started, graph ID %s, partition ID %s, epochs %s',args['graph_id'],args['partition_id'],args['epochs'])
+
+    path_nodes = args['path_nodes'] + args['graph_id'] + '_nodes_' + args['partition_id'] + ".csv"
+    nodes = pd.read_csv(path_nodes,index_col=0)
+    #nodes = nodes.astype("float32")
+
+    path_edges = args['path_edges'] + args['graph_id'] + '_edges_' + args['partition_id'] + ".csv"
+    edges = pd.read_csv(path_edges)
+    #edges = edges.astype({"source":"uint32","target":"uint32"})
+
+    logging.info('Model initialized')
+    model = Model(nodes,edges)
+    num_train_ex,num_test_ex = model.initialize()
+
+    graph_params = (num_train_ex,num_test_ex)
+
+    logging.info('Number of training examples - %s, Number of testing examples %s',num_train_ex,num_test_ex)
+
+    client = Client(model,graph_params,weights_path=args['path_weights'],graph_id=args['graph_id'],partition_id=args['partition_id'],epochs = int(args['epochs']) ,IP=args['IP'],PORT=int(args['PORT']))
+
+
+    logging.info('Federated training started!')
+
+    start = timer()
+    client.run()
+    end = timer()
+
+    elapsed_time = end -start
+    logging.info('Federated training done!')
+    logging.info('Training report : Elapsed time %s seconds, graph ID %s, partition ID %s, epochs %s',elapsed_time,args['graph_id'],args['partition_id'],args['epochs'])
+
+    
```

Comment:
Lets keep only one empty line at the bottom of each Python file

Reply:
done

Reply:
Ok